tests/testthat/test-opart-gaussian.R

```diff
@@ -60,3 +84,21 @@ res <- opart_gaussian(selProfile$logratio, 0)
 test_that("all the data points are segment ends", {
   expect_equal(nrow(as.data.frame(res$end.vec)), nrows - 2)
 })
+
+
+#tests for opart_poisson
+
+test_that("opart gives error for negative penalty", {
+  expect_error({
+    opart_poisson(c(1,2,3), -2)
+  }, "penalty value must be greater than 0",
+  fixed=TRUE)
+})
+
+
+test_that("opart gives error when data points are less than 1", {
```

Comment:
when the data vector length is less than one

(data points/values can be less than 1, for example 0 is totally fine for the Poisson distribution)